sycl/doc/extensions/experimental/sycl_ext_oneapi_enqueue_native_command.asciidoc

```diff
@@ -0,0 +1,216 @@
+      // This stream has been synchronized with dep_event's underlying
+      // hipEvent_t
+      hipStream_t stream = h.get_native_queue<sycl::backend::hip>();
+      hipMemcpyAsync(target_ptr, native_mem, test_size * sizeof(int),
+                      hipMemcpyDeviceToHost, stream);
+    });
+  });
+q.wait()
+```
+
+The print statement may print before `dep_event` has completed. However, the
+asynchronous memcpy submitted to the native queue obtained by
+`interop_handle::get_native_queue` is guaranteed to have the correct
+dependencies, and therefore will only start once its dependent events have
+completed.
+
+By contrast, when using a `host_task`, it is guaranteed that the print statement
+will only happen once the host task's dependent events are observed to be
+complete on the host.
+
+A SYCL event returned by a submission of a
+`ext_oneapi_enqueue_native_command` command is only complete once the
+asynchronous work enqueued to the native queue obtained through
+`interop_handle::get_native_queue()` has completed.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_ENQUEUE_NATIVE_COMMAND` to one of the values defined
+in the table below.  Applications can test for the existence of this macro to
+determine if the implementation supports this feature, or applications can test
+the macro's value to determine which of the extension's features the
+implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== Additions to handler class
+
+This extension adds the following new member function to the SYCL `handler`
+class:
+
+```c++
+class handler {
+  template <typename Func>
+  void ext_oneapi_enqueue_native_command(Func&& interopCallable);
+};
+```
+
+_Constraints_: The `Func` must a C++ callable object which takes a single
+parameter of type `interop_handle`. `interopCallable` must only submit
+asynchronous interop work to the backend, through the native backend object
+obtained from `interop_handle::get_native_queue`.
+
+_Effects_: The `interopCallable` object is called exactly once, and this call
+may be made asynchronously even after the calling thread returns from
+`ext_oneapi_enqueue_native_command`.
+
+The call to `interopCallable` may submit one or more asynchronous tasks to the
+native backend object obtained from `interop_handle::get_native_queue`, and
+these tasks become encapsulated in a SYCL command that is added to the queue.
+If the enclosing command group has any dependencies, these dependencies will be
+propagated to the native API during the command submission, meaning no further
+dependency analysis is needed within the native API. This happens, for example,
```

Comment:
I think the main subtle point of this extension is that you don't need to handle dependencies in the native API in `interopCallable`. I think this is worth spelling out. I have reworded this slightly to make it a bit clearer that deps don't need to be managed by the user in `interopCallable`

Reply:
The wording "no further dependency analysis is needed" seems pretty vague to me.  I also think we should reorder the sentences, so that "This happens ..." is closer to sentence that "this" refers to.  Finally, I think we need to reword "the SYCL command" because we removed the previous sentence that defined what "command" we are talking about.

Putting this all together:

> If the enclosing command group has any dependencies, these dependencies are propagated to the native asynchronous tasks.  This happens, for example, if the command group calls `handler::depends_on` or if it constructs an `accessor`.  As a result, there is typically no need to specify these dependencies through native APIs.  Note, however, that these dependencies are associated with the *asynchronous tasks submitted by* `interopCallable`, not the call to `interopCallable`. The call to `interopCallable` may happen even before the dependencies are satisfied.

Reply:
I'm happy with that wording. I've updated the docs with this exact wording.